ejb3/src/main/java/org/jboss/as/ejb3/concurrency/ContainerManagedConcurrencyInterceptor.java

```diff
@@ -70,7 +69,13 @@ public Object processInvocation(final InterceptorContext context) throws Excepti
         if (method == null) {
             throw EjbLogger.ROOT_LOGGER.invocationNotApplicableForMethodInvocation(invocationContext);
         }
-        Method invokedMethod = viewMethodToComponentMethodMap.get(method);
+        Method invokedMethod = null;
+        for (Map<Method, Method> methodMethodMap : ((SingletonComponent) lockableComponent).getViewMethodToComponentMethodMapList()) {
```

Comment:
This cast doesn't look a good idea due to the high coupling between the interceptor and the singleton. Maybe the getter could be moved to the lockeableComponent instead or maybe the issue must be resolved with a different solution for the issue, any suggestion?